src/bridger/Bridger.sol

```diff
@@ -241,81 +257,103 @@ contract Bridger is
-        address _kintoWallet,
-        address _inputAsset,
-        address _finalAsset,
-        uint256 _amount,
-        uint256 _minReceive,
-        SwapData calldata _swapData
-    ) private {
-        uint256 amountBought = _amount;
-        if (_inputAsset != _finalAsset) {
-            if (_inputAsset == ETH && _finalAsset == wstETH) {
-                amountBought = _swapETHtoWstETH(_amount);
-            } else {
-                if (_inputAsset == ETH) {
-                    _amount = _amount - _swapData.gasFee;
-                    WETH.deposit{value: _amount}();
-                    _inputAsset = address(WETH);
-                }
-                amountBought = _executeSwap(_inputAsset, _finalAsset, _amount, _minReceive, _swapData);
-            }
-
-            if (_finalAsset == sUSDe) {
-                amountBought = _stakeAssets(USDe, amountBought);
-            }
+    function _deposit(
+        address user,
+        address inputAsset,
+        uint256 amount,
+        address kintoWallet,
+        address finalAsset,
+        uint256 minReceive,
+        bytes calldata swapCallData,
+        BridgeData calldata bridgeData
+    ) internal {
+        if (amount == 0) revert InvalidAmount(0);
+
+        _checkFinalAsset(finalAsset);
+
+        if (inputAsset != finalAsset && !allowedAssets[inputAsset]) {
+            revert InvalidInputAsset(inputAsset);
         }
 
-        depositCount++;
-        emit Deposit(_sender, _kintoWallet, _inputAsset, _amount, _finalAsset, amountBought, depositCount);
+        // slither-disable-next-line arbitrary-send-erc20
+        IERC20(inputAsset).safeTransferFrom(user, address(this), amount);
+
+        uint256 amountBought = _swap(inputAsset, finalAsset, amount, minReceive, swapCallData);
+
+        // Bridge the final amount to Kinto
+        IERC20(finalAsset).safeIncreaseAllowance(bridgeData.vault, amountBought);
+        IBridge(bridgeData.vault).bridge{value: bridgeData.gasFee}(
+            kintoWallet,
+            amountBought,
+            bridgeData.msgGasLimit,
+            bridgeData.connector,
+            bridgeData.execPayload,
+            bridgeData.options
+        );
+
+        emit Deposit(user, kintoWallet, inputAsset, amount, finalAsset, amountBought);
     }
 
-    function _executeSwap(
-        address _inputAsset,
-        address _finalAsset,
-        uint256 _amount,
-        uint256 _minReceive,
-        SwapData calldata _swapData
+    function _swap(
+        address inputAsset,
+        address finalAsset,
+        uint256 amount,
+        uint256 minReceive,
+        bytes calldata swapCallData
     ) private returns (uint256 amountBought) {
-        amountBought = _amount;
-        if (_finalAsset == sUSDe) _finalAsset = USDe; // if sUSDE, swap to USDe & then stake
-        if (_finalAsset != _inputAsset) {
-            if (!swapsEnabled) revert SwapsDisabled();
+        amountBought = amount;
+        if (inputAsset != finalAsset) {
```

Comment:
```suggestion
        if (inputAsset == finalAsset) {
```

Reply:
That is why need a fork tests for the swap. Lmao. I was wondering how these tests work.